### PR TITLE
Some Client Simulator reworks

### PR DESCRIPTION
**Pull request by dawagner**
* Remove some **dead code** and simplify some code portions.
* Add a **new feature**: pass more than one '--scenario' option.
* **Remove the `SetupScript` config item** (if you need a setup script, you need to run it by yourself)
* The `TestPlatformHost` and `ParameterFrameworkHost` config items used to contain both the hosts and the ports; **replace them** with `TestPlatformPort` and `ParameterFramworkPort` that contain only the port. The host wil always be `localhost` anyway - no need to define it.
* **Remove the `RemoteProcessCommand` and `TestPlatformCommand` config items** and expect `remote-process` and `test-platform` to be in the PATH.
* Make some config items optional.
* Make all paths in conf.json relative to the test directory (but obsolute paths will really be considered absolute).

### CODE REVIEW

File: tools/clientSimulator/README.md [hunk 1]
```diff
@@ -116,12 +109,8 @@ working directory*.
...
     "LogFile" : "tests.log",
 
-    "SetupScript" : "echo 'bouh'",
-
-    "TestPlatformCommand" : "test-platform",
-    "RemoteProcessCommand" : "remote-process",
-    "TestPlatformHost" : "localhost 5001",
-    "ParameterFramworkHost" : "localhost 5000",
+    "TestPlatformPort" : "5001",
+    "ParameterFramworkPort" : "5000",
```
Comment (dpertuze) [inline tools/clientSimulator/README.md]: Maybe the typo should be corrected : Framwork/Framework

**Comment by dawagner** [comment]
@dpertuze the "ParameterFramworkPort" config item wasn't even used... So I removed it and pushed a new version. Thanks for catching it.

File: tools/clientSimulator/pfClientSimulator.py [hunk 2]
```diff
@@ -175,34 +185,33 @@ def main():
...
+                     ))
+                for scenarioFileName in sorted(os.listdir(configParser["ScenariosDirectory"]))
+            ]
+            if args.scenario is not None:
+                for elem in args.scenario:
+                    scenarioOptions[elem][1]()
+            if args.no_exit:
+                # Let the user choose more scenarios after the ones chosen by command line
+                # or if none was given on the command line.
+                UserInteractor.getMenu(scenarioOptions, "Quit")
```
Comment (dpertuze) [inline tools/clientSimulator/pfClientSimulator.py]: In this actual version, if the user wants to get to the menu, no_exit option must be put on the command line, which does not seem natural for me...

**Comment by dawagner** [comment]
@dpertuze done.

**Comment by clero** [comment]
:+1:

**Comment by sguiriec** [comment]
Tested-by: Sebastien Guiriec <sebastien.guiriec@intel.com>

**Pull request merged**